Detectors/TPC/reconstruction/src/KrBoxClusterFinder.cxx

```diff
@@ -0,0 +1,250 @@
+  for (int iRow = 0; iRow < mMaxRows; iRow++) {
+    for (int iPad = 0; iPad < mMaxPads; iPad++) {
+      for (int iTime = 0; iTime < mMaxTimes; iTime++) {
+
+        mTempCluster.totCharge = 0;
+        mTempCluster.maxCharge = 0;
+        mTempCluster.size = 0;
+        mTempCluster.meanPad = 0;
+        mTempCluster.meanRow = 0;
+        mTempCluster.meanTime = 0;
+        mTempCluster.sigmaPad = 0;
+        mTempCluster.sigmaRow = 0;
+        mTempCluster.sigmaTime = 0;
+        int qMax = mMapOfAllDigits.at(iTime).at(iRow).at(iPad);
+
+        // cluster Maximum must at least be larger than Threshold
+        if (qMax <= mQThresholdMax) {
+          continue;
+        }
+
+        // Acceptance condition: Require at least mMinNumberOfNeighbours neigbours
+        // with signal in any direction!
+        int noNeighbours = 0;
+        if ((iPad + 1 < mMaxPads &&
+             mMapOfAllDigits.at(iTime).at(iRow).at(iPad + 1) > mQThreshold))
+          noNeighbours++;
+        if ((iPad - 1 >= 0 &&
+             mMapOfAllDigits.at(iTime).at(iRow).at(iPad - 1) > mQThreshold)) {
+          noNeighbours++;
+        }
+        if ((iTime + 1 < mMaxTimes &&
+             mMapOfAllDigits.at(iTime + 1).at(iRow).at(iPad) > mQThreshold))
+          noNeighbours++;
+        if ((iTime - 1 >= 0 &&
+             mMapOfAllDigits.at(iTime - 1).at(iRow).at(iPad) > mQThreshold)) {
+          noNeighbours++;
+        }
+        if ((iRow + 1 < mMaxRows &&
+             mMapOfAllDigits.at(iTime).at(iRow + 1).at(iPad) > mQThreshold))
+          noNeighbours++;
+        if ((iRow - 1 >= 0 &&
+             mMapOfAllDigits.at(iTime).at(iRow - 1).at(iPad) > mQThreshold)) {
+          noNeighbours++;
+        }
+        if (noNeighbours < mMinNumberOfNeighbours) {
+          continue;
+        }
+
+        // Check that this is a local maximum
+        // Note that the checking is done so that if 2 charges have the same
+        // qMax then only 1 cluster is generated
+        // (that is why there is BOTH > and >=)
+        // -> only the maximum with the smalest indices will be accepted
+        bool thisIsMax = true;
+
+        for (int i = -mMaxClusterSizePad; i <= mMaxClusterSizePad && thisIsMax;
+             i++) {
+          for (int k = -mMaxClusterSizeRow; k <= mMaxClusterSizeRow && thisIsMax;
+               k++) {
+            for (int j = -mMaxClusterSizeTime;
+                 j <= mMaxClusterSizeTime && thisIsMax; j++) {
+              if (iPad + i < mMaxPads && iTime + j < mMaxTimes &&
+                  iRow + k < mMaxRows && iPad + i >= 0 && iTime + j >= 0 &&
+                  iRow + k >= 0 &&
+                  mMapOfAllDigits.at(iTime + j).at(iRow + k).at(iPad + i) > qMax) {
+                thisIsMax = false;
+              }
+            }
+          }
+        }
+
+        if (!thisIsMax) {
+          continue;
+        } else {
+          localMaximaCoords.push_back(std::make_tuple(iPad, iRow, iTime));

```

Comment:
can you try
```c++
localMaximaCoords.emplace_back({iPad, iRow, iTime});
```

Reply:
It didn't compile.

Reply:
It did not compile.

Reply:
Still you should be able to use `emplace_back(std::make_tuple(iPad, iRow, iTime));`